### PR TITLE
boards: Remove OUTPUT_FORMAT and OUTPUT_ARCH from ld script

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
let toolchain decide the correct value base on the command line

Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>
Change-Id: I342db6a88e4a161a322a8fea48a59e6ca7617ae6

### CODE REVIEW

File: boards/arm/tms570/launchxl-tms57004/scripts/flash-sram.ld [hunk 1]
```diff
@@ -43,8 +43,6 @@ MEMORY
   sram (rwx) : ORIGIN = 0x08000000, LENGTH = 32K
 }
 
-OUTPUT_ARCH(arm)
-OUTPUT_FORMAT(elf32-bigarm)
```
Comment (patacongo) [inline boards/arm/tms570/launchxl-tms57004/scripts/flash-sram.ld:47]: @xiaoxiang781216  @liuguo09  Notice that ARM big-endian is required for the TMS470 family.  Does this change break TMS570?

File: boards/arm/tms570/tms570ls31x-usb-kit/scripts/flash-sram.ld [hunk 2]
```diff
@@ -45,8 +45,6 @@ MEMORY
   sram (rwx) : ORIGIN = 0x08000000, LENGTH = (0x0003EB00 + 0x00001500)
 }
 
-OUTPUT_ARCH(arm)
-OUTPUT_FORMAT(elf32-bigarm)
```
Comment (patacongo) [inline boards/arm/tms570/tms570ls31x-usb-kit/scripts/flash-sram.ld:49]: See above

File: boards/arm/tms570/launchxl-tms57004/scripts/flash-sram.ld (hunk 1 above)
Comment (patacongo) [reply boards/arm/tms570/launchxl-tms57004/scripts/flash-sram.ld:47]: I tried to build a TMS570 configuration, but it needs an armeb-nuttx-eabi-gcc toolchain and don't have one right now.

Comment (xiaoxiang781216) [reply boards/arm/tms570/launchxl-tms57004/scripts/flash-sram.ld:47]: I suppose the toolchain can select the right endian base on the command line option if the linker script file doesn't specify one explicitly.

Comment (patacongo) [reply boards/arm/tms570/launchxl-tms57004/scripts/flash-sram.ld:47]: > 
> 
> I suppose the toolchain can select the right endian base on the command line option if the linker script file doesn't specify one explicitly.

All of these untested changes to code is irresponsible and unprofessional.

**Pull request merged**